deflate.c

```diff
@@ -1052,8 +1053,6 @@ int32_t ZEXPORT PREFIX(deflate)(PREFIX3(stream) *strm, int32_t flush) {
 
     if (flush != Z_FINISH)
         return Z_OK;
-    if (s->wrap <= 0)
-        return Z_STREAM_END;
```

Comment:
This and all the below changes to deflate.c look like they could change compressed output, and need to be tested for that. Have you observed any changes in output on non-DFLTCC builds?
If this results in any changes, we need to understand them completely, to make sure they do not cause regressions on certain input data.

Reply:
No, this should not change anything for non-DFLTCC builds, and I did not observe changes in compressed data.

wrap can be -1/-2 ("done"), 0 (raw), 1 (zlib) or 2 (gzip).
For wrap values 1 and 2 everything stays exactly as it used to be.
For -1/-2 and 0 (wrap == 0 is the reason why this is needed), there is now an additional `flush_pending(strm)` call. In non-DFLTCC builds there is nothing to flush, so this is a no-op.